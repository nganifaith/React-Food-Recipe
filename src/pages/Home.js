import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setMeals } from '../actions';
import { searchMeals } from '../API';
import { getMeals } from '../reducers/mealsList';

import Main from '../components/Main';
import SearchBar from '../components/SearchBar';
import MealImage from '../components/MealImage';

const Home = () => {
  const [searchTerm, setSearchTerm] = useState('');
  const dispatch = useDispatch();
  const meals = useSelector(getMeals);
  useEffect(() => {
    searchMeals(searchTerm)
      .then(({ meals }) => {
        dispatch(setMeals(meals));
      })
      .catch((e) => {
        console.error(e);
      });
  }, [searchTerm]);
  return (
    <div>
      <SearchBar setSearchTerm={setSearchTerm} />
      <Main header={searchTerm ? 'Search Results' : 'All Meals'}>
        {meals.map((meal) => (
          <MealImage key={meal.idMeal} meals={meal} />
        ))}
      </Main>
    </div>
  );
};

export default Home;
