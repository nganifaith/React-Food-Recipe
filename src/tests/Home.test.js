import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducer from '../reducers';

import Home from '../pages/Home';

const store = createStore(rootReducer);

it('renders correctly', () => {
  const home = renderer
    .create(
      <Provider store={store}>
        <Home />
      </Provider>,
    )
    .toJSON();
  expect(home).toMatchSnapshot();
});

it('renders banner', () => {
  render(
    <Provider store={store}>
      <Home />
    </Provider>,
  );
  const searchIcon = screen.getByRole('img');
  expect(searchIcon).toBeInTheDocument();
});

it('renders banner', () => {
  render(
    <Provider store={store}>
      <Home />
    </Provider>,
  );
  const searchBar = screen.getByRole('textbox');
  expect(searchBar).toBeInTheDocument();
});

it('renders banner', () => {
  render(
    <Provider store={store}>
      <Home />
    </Provider>,
  );
  const mealSection = screen.getByRole('heading');
  expect(mealSection).toBeInTheDocument();
});
