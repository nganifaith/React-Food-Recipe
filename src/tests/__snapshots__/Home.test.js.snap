// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="sc-dlfnuX iBRhvT"
  >
    <div
      className="sc-hKgJUU fIcSxm"
    >
      <img
        alt="search-icon"
        src="search-icon.svg"
      />
      <input
        onChange={[Function]}
        placeholder="Search A Meal"
        type="text"
      />
    </div>
  </div>
  <span>
    
  </span>
  <div
    className="sc-bdfBQB jWmjXB"
  >
    <h1>
      All Meals
    </h1>
    <div
      className="sc-gsTEea fLetFe"
    />
  </div>
</div>
`;
